Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -1597,6 +1597,21 @@ namespace Tensile
         assert(pAMDGPU != nullptr && pAMDGPU->computeUnitCount != 0);
         size_t cuCount = pAMDGPU->computeUnitCount;
         size_t skGrid  = cuCount;
+        if(pAMDGPU->skDynamicGrid == 2 && tiles > skGrid)
```

Comment:
Any good location to put some explanation about how TENSILE_STREAMK_DYNAMIC_GRID works?

Reply:
Probably, better to put some brief explanation about TENSILE_STREAMK_DYNAMIC_GRID

Reply:
Does this new logic work with skGridMultiplier?

Reply:
Posted an update to add a comment to Common.py alongside the documentation for the other STREAMK environment settings. The comment is brief for now since the exact behaviour and usage is likely to change a bit as it gets more test data collected. But now there's a high-level description and a note on how to use it.
The new logic can also work with the GridMultiplier. I'd guess that combing the two would not be the optimal way to run stream-k, but we need to run more tests to see if there is a use-case for it.

Reply:
Regarding GridMultiplier, what I am thinking is to apply this before calculating reduced grid.
Probably, we do not need to combine them.
You can think about that if needed.